WalletWasabi.Fluent/Behaviors/HistoryItemDetailsBehavior.cs

```diff
@@ -0,0 +1,118 @@
+using System.Reactive.Disposables;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Threading;
+using DataBox;
+
+namespace WalletWasabi.Fluent.Behaviors
+{
+	public class HistoryItemDetailsBehavior : DisposingBehavior<DataBoxRow>
+	{
+		private HistoryItemDetailsAdorner? _historyItemDetailsAdorner;
+		private IDisposable? _currentAdornerEvents;
+
+		protected override void OnAttached(CompositeDisposable disposables)
+		{
+			if (AssociatedObject is not null)
+			{
+				disposables.Add(
+					AssociatedObject
+						.GetObservable(InputElement.IsPointerOverProperty)
+						.Subscribe(x =>
+						{
+							if (AssociatedObject.IsPointerOver)
+							{
+								AddAdorner(AssociatedObject);
+							}
+							else
+							{
+								CheckIfShouldRemove();
+							}
+						}));
+
+				disposables.Add(
+					AssociatedObject
+						.GetObservable(ListBoxItem.IsSelectedProperty)
+						.Subscribe(x =>
+						{
+							if (AssociatedObject.IsSelected)
+							{
+								AddAdorner(AssociatedObject);
+							}
+							else
+							{
+								CheckIfShouldRemove();
+							}
+						}));
+			}
+		}
+
+		private void CheckIfShouldRemove()
+		{
+			Dispatcher.UIThread.Post(() =>
+			{
+				if (_historyItemDetailsAdorner != null && !AssociatedObject.IsPointerOver &&
```

Comment:
Check for AssociatedObject is not null here, when switching wallet this can get null